downgrade.jl

```diff
@@ -8,6 +8,277 @@ julia_version = length(ARGS) >= 4 ? ARGS[4] : "1.10"
 valid_modes = ["deps", "alldeps", "weakdeps", "forcedeps"]
 mode in valid_modes || error("mode must be one of: $(join(valid_modes, ", "))")
 
+"""
+    get_local_source_packages(project_file)
+
+Parse a Project.toml and find packages that have local path sources.
+Returns a Set of package names that should be excluded from resolution
+because they are sourced from local paths (e.g., the main package in test/Project.toml).
+
+In Julia 1.13+, test dependencies often use [sources.PackageName] with path=".."
+to reference the main package. These cannot be resolved from the registry.
+"""
+function get_local_source_packages(project_file::String)
+    local_pkgs = Set{String}()
+
+    if !isfile(project_file)
+        return local_pkgs
+    end
+
+    project = TOML.parsefile(project_file)
+
+    # Check for [sources] section entries with path keys
+    sources = get(project, "sources", Dict())
+    for (pkg_name, source_info) in sources
+        if source_info isa Dict && haskey(source_info, "path")
+            push!(local_pkgs, pkg_name)
+            @info "Found local source package: $pkg_name (path=$(source_info["path"]))"
```

Comment:
Do we want this level of verbosity in the final PR?

Reply:
I don't think it's particularly bad to have more printed here